File: app/src/main/java/co/com/ceiba/mobile/pruebadeingreso/view/adapters/UsersAdapter.java
package co.com.ceiba.mobile.pruebadeingreso.view.adapters;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import co.com.ceiba.mobile.pruebadeingreso.R;
import co.com.ceiba.mobile.pruebadeingreso.storage.models.User;

public class UsersAdapter extends RecyclerView.Adapter<UsersAdapter.UserViewHolder> {

    List<User> users = new ArrayList<>();
    OnUserPostListener userListener;

    public interface OnUserPostListener {
        void onClick(User user);
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.user_list_item, parent, false);

        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        User user = users.get(position);

        holder.name.setText(user.name);
        holder.phone.setText(user.phone);
        holder.email.setText(user.email);
        holder.viewPost.setOnClickListener(view -> {
            userListener.onClick(user);
        });
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    public void setUsers(List<User> users) {
        this.users = users;
        notifyDataSetChanged();
    }

    public void setUserListener(OnUserPostListener userListener) {
        this.userListener = userListener;
    }

    static class UserViewHolder extends RecyclerView.ViewHolder {

        public TextView name;
        public TextView phone;
        public TextView email;
        public Button viewPost;

        public UserViewHolder(View view) {
            super(view);

            name = view.findViewById(R.id.name);
            phone = view.findViewById(R.id.phone);
            email = view.findViewById(R.id.email);
            viewPost = view.findViewById(R.id.btn_view_post);
        }
    }

}

File: app/src/main/java/co/com/ceiba/mobile/pruebadeingreso/presentation/adapters/UsersAdapter.kt
package co.com.ceiba.mobile.pruebadeingreso.presentation.adapters

import android.annotation.SuppressLint
import androidx.recyclerview.widget.RecyclerView
import co.com.ceiba.mobile.pruebadeingreso.presentation.adapters.UsersAdapter.UserViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import co.com.ceiba.mobile.pruebadeingreso.R
import android.widget.TextView
import co.com.ceiba.mobile.pruebadeingreso.storage.models.User

class UsersAdapter(
    private var users: List<User> = emptyList(),
    private var userListener: ((user: User) -> Unit)
) : RecyclerView.Adapter<UserViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val view = LayoutInflater
            .from(parent.context)
            .inflate(R.layout.user_list_item, parent, false)
        return UserViewHolder(view)
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        val user = users[position]
        holder.name.text = user.name
        holder.phone.text = user.phone
        holder.email.text = user.email
        holder.viewPost.setOnClickListener { userListener(user) }
    }

    override fun getItemCount(): Int {
        return users.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setUsers(users: List<User>) {
        this.users = users
        notifyDataSetChanged()
    }

    class UserViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var name: TextView = view.findViewById(R.id.name)
        var phone: TextView = view.findViewById(R.id.phone)
        var email: TextView = view.findViewById(R.id.email)
        var viewPost: Button = view.findViewById(R.id.btn_view_post)
    }
}